response_operations_ui/static/js/validate-ci.js

```diff
@@ -1,50 +1,48 @@
 
-            errorPanelBody.classList.remove('panel__body');
+            if (!Array.isArray(classes)) throw new Error('Expected Array as second argument');
 
-            // Create array from NodeList
-            paragraphsToRemoveClassFrom = Array.prototype.slice.call(errorPanelBody.querySelectorAll('p'));
+            if (['add', 'remove'].indexOf(action) === -1) throw new Error('Expected add or remove as third parameter');
 
-            // Iterate array of DOM elements to remove hidden class from child paragraphs
-            paragraphsToRemoveClassFrom.forEach(function(element) {
-                element.classList.add('hidden');
-            });
+            classes.map(c => node.classList[action](c));
+        },
 
-            document.getElementById('btn-load-ci').classList.remove('unready');
-        } else {
-            errorPanel.classList.add('panel');
-            errorPanel.classList.add('panel--simple');
-            errorPanel.classList.add('panel--error');
+        arrayLikeToArray: function arrayLikeToArray(arrayLike) {
+            if (!('length' in arrayLike) && !('size' in arrayLike)) throw new Error('Expected array-like object as argument');
+
+            return Array.prototype.slice.call(arrayLike);
+        },
+    },
+
+    checkCI: function checkCI(file) {
+        const type = file.type;
+        const errorPanel = document.getElementById('ciFileErrorPanel');
+        const errorPanelBody = document.getElementById('ciFileErrorPanelBody');
+        const button = document.getElementById('btn-load-ci');
 
-            errorPanelBody.classList.add('panel__body');
-            paragraphsToAddClassTo = Array.prototype.slice.call(errorPanelBody.querySelectorAll('p'));
+        const fileIsODS = type === 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet';
+        const mainAction = fileIsODS ? 'remove' : 'add';
+        const contraAction = fileIsODS ? 'add' : 'remove';
 
-            paragraphsToAddClassTo.forEach(function(element) {
-                element.classList.remove('hidden');
-            });
+        this.__private__.nodeClassesChange(errorPanel, ['panel', 'panel--simple', 'panel--error'], mainAction);
+        this.__private__.nodeClassesChange(errorPanelBody, ['panel__body'], mainAction);
+        this.__private__.nodeClassesChange(button, ['unready'], contraAction);
 
-            document.getElementById('btn-load-ci').classList.add('unready');
-        }
-    }
+        this.__private__.arrayLikeToArray(errorPanelBody.querySelectorAll('p')).forEach(el => {
+            this.__private__.nodeClassesChange(el, ['hidden'], contraAction);
+        });
+    },
 
-    function checkSelectedCI(files) {
+    checkSelectedCI: function checkSelectedCI(files) {
         // Check for the various File API support.
         if (window.FileReader) {
-            // FileReader are supported.
-            checkCI(files[0]);
+            // FileReader is supported.
```

Comment:
+1 for grammar

Reply:
Yeah, I don't know where that originally came from, but I had to change it.